elyra/pipeline/validate.py

```diff
@@ -249,7 +251,7 @@ async def _validate_node_properties(self, pipeline: dict, pipeline_runtime: str,
             components = ComponentRegistry.to_canvas_palette(component_list, categories)
             for node in node_list:
                 if node['type'] == 'execution_node':
-                    if self._is_legacy_pipeline(pipeline):
+                    if self._is_legacy_pipeline(pipeline) and not self._is_standalone_submission(pipeline_source):
```

Comment:
Before this change, only a "legacy pipeline" could get into this block.  Now a legacy pipeline can fall into the else block if it happens to be a "standalone submission" - which seems a bit odd, but I guess that's the issue being resolved, although this _also_ changes where the `node_data` is derived relative to standalone submissions.  Was that also an issue?

Might it be less confusing to determine `node_label`, for all invocation "styles", by first looking at `['ui_data']['labe'l']` (and tolerating `ui_data`'s absence, then (and only then) fallback to `['app_data']['label']` since `ui_data` is the first source of truth and will not be present in standalone submissions.

```python
if node_data['ui_data']:  # If present, always the source of truth
    node_label = node_data['ui_data'].get('label')
else:
    node_label = node['app_data'].get('label')
```
Also, shouldn't the label always have a value and should that be checked here (or soon) as part of the validation?

Reply:
thanks for the review, I will make the updates shortly.
Need to better organize the scenarios..=|

Reply:
moving the standalone submission to its own block. 
adding the label logic to the legacy conditional block...weird thing...in a migrated pipeline from 3->4, are we allowed to have two label fields in the node? 
![image](https://user-images.githubusercontent.com/13340013/127954746-b2ce9644-a965-4a82-9882-5749529fa91e.png)

Reply:
Yes, this is possible.  This is why `['app_data']['ui_data']['label']` is the _first_ source of truth as that will always contain the correct value - when it exists.  However, there are some circumstances (like "standalone submission") when it won't exist, in which case `['app_data']['label']` should be used.  My comment was to not look at the "submission style" and always reach for `['app_data']['ui_data']['label']` while tolerating its absence.